src/jobs/pauseAppCheck.ts

```diff
@@ -0,0 +1,191 @@
+// Job to check for applications that have reached attestationBy date and are not attested
+// These will be PAUSED and notifications sent
+async function runPauseAppsCheck(
+  currentDate: Date,
+  emailClient: Transporter<SMTPTransport.SentMessageInfo>,
+  user: Identity,
+): Promise<JobReport<BatchJobDetails>> {
+  const jobStartTime = new Date();
+  try {
+    logger.info(`${JOB_NAME} - Initiating...`);
+    const details = await getPausedAppsReportDetails(emailClient, user, currentDate);
+    details.errors.length
+      ? logger.info(`${JOB_NAME} - Completed.`)
+      : logger.warn(`${JOB_NAME} - Completed with errors.`);
+    const endTime = new Date();
+    const jobSuccessReport: JobReport<BatchJobDetails> = {
+      jobName: JOB_NAME,
+      startedAt: jobStartTime,
+      finishedAt: endTime,
+      success: true,
+      details,
+    };
+    logger.info(`${JOB_NAME} - Report: ${JSON.stringify(jobSuccessReport)}`);
+    return jobSuccessReport;
+  } catch (err) {
+    logger.error(`${JOB_NAME} - Failed to complete, with error: ${(err as Error).message}`);
+    const jobEndTime = new Date();
+    const jobFailedReport: JobReport<BatchJobDetails> = {
+      jobName: JOB_NAME,
+      startedAt: jobStartTime,
+      finishedAt: jobEndTime,
+      success: false,
+      error: `${JOB_NAME} - Failed to complete, with error: ${(err as Error).message}`,
+    };
+    logger.error(`${JOB_NAME} - Report: ${JSON.stringify(jobFailedReport)}`);
+    return jobFailedReport;
+  }
+}
+
+const searchPauseableApplications = async (
+  query: FilterQuery<ApplicationDocument>,
+): Promise<ApplicationDocument[]> => {
+  // can still run this multiple times in the same 24hr period,
+  // as any apps that were paused previously will be ignored because we're looking for APPROVED state
+  const apps = await ApplicationModel.find(query).exec();
+  return apps;
+};
+
+// update app state, including transition to paused + update event
+const pauseApplication = async (
+  currentApp: Application,
+  identity: Identity,
+  reason?: PauseReason,
+): Promise<Application> => {
+  const config = await getAppConfig();
+  // set app in state
+  const appObj = new ApplicationStateManager(currentApp, config);
+  const role = await getDacoRole(identity);
+  logger.info(
+    `${JOB_NAME} - Role ${role} is trying to PAUSE appId ${currentApp.appId} with pause reason ${reason}`,
+  );
+  const result = appObj.updateApp({ state: 'PAUSED', pauseReason: reason }, false, {
+    id: identity.userId,
+    role,
+  });
+  logger.info(`${JOB_NAME} - Updating ${result.appId} in db to ${result.state}.`);
+  // save new app state in db
+  await ApplicationModel.updateOne({ appId: result.appId }, result);
+  // retrieve updated app from db
+  const updatedApp = await ApplicationModel.findOne({
+    appId: result.appId,
+  }).exec();
+  if (updatedApp) {
+    logger.info(`${JOB_NAME} - Returning updated app ${updatedApp?.appId}.`);
+    return updatedApp?.toObject();
+  }
+  logger.error(
```

Comment:
this check i'm not sure about - I was trying to get rid of the `return updatedApp.toObject() as Application` typing i had previously, but returning the inputted app seems maybe not ideal because the state change could have been successful even if for some reason the `findOne` fails...thoughts?